rules-reviewed/openshift/socket-communication.windup.xml

```diff
@@ -56,30 +60,18 @@
         <rule id="socket-communication-00002">
             <when>
                 <or>
-                  <javaclass references="java.net.ServerSocket{*}">
-                      <location>CONSTRUCTOR_CALL</location>
-                      <location>IMPORT</location>
-                      <location>INHERITANCE</location>
-                      <location>METHOD_CALL</location>
-                      <location>VARIABLE_DECLARATION</location>
-                  </javaclass>
-                    <javaclass references="java.nio.channels.AsynchronousServerSocketChannel{*}">
-                        <location>CONSTRUCTOR_CALL</location>
-                        <location>IMPORT</location>
-                        <location>VARIABLE_DECLARATION</location>
-                    </javaclass>
-                    <javaclass references="java.nio.channels.ServerSocketChannel{*}">
-                        <location>CONSTRUCTOR_CALL</location>
-                        <location>IMPORT</location>
-                        <location>VARIABLE_DECLARATION</location>
-                    </javaclass>
+                  <javaclass references="java.net.ServerSocket{*}"/>
+                  <javaclass references="java.nio.channels.AsynchronousServerSocketChannel{*}"/>
+                  <javaclass references="java.nio.channels.ServerSocketChannel{*}"/>
                 </or>
             </when>
             <perform>
-                <hint title="Java APIs for server sockets" effort="3" category-id="cloud-mandatory">
+                <hint title="Server sockets communication" effort="5" category-id="cloud-optional">
                     <message>
-                      ServerSockets are not allowed in Java EE. Consider using Java EE standard communication features, especially loosely coupled protocols (JAW-WS, JAX-RS) instead.  
+                      Direct communication through sockets and channels is an anti-pattern in cloud environment because it isn't a reliable and scalable way of interact with other systems.  
```

Comment:
Here I would add one sentence explaining at the beginning:

"Java sockets are internal endpoints of two-way communications. They are defined by an IP address, port, and protocol (TCP/UDP)."